dwave/optimization/src/nodes/mathematical.cpp

```diff
@@ -411,8 +513,38 @@ double NaryOpNode<BinaryOp>::min() const {
     if constexpr (std::is_same<result_type, bool>::value) {
         return false;
     }
-    // there are other cases we could/should handle here.
-    return Array::min();
+
+    // these can result in inf. If we update propagation/initialization to handle
+    // that case we should update these as well.
+    if constexpr (std::is_same<BinaryOp, std::plus<double>>::value) {
+        return std::accumulate(operands_.begin(), operands_.end(), 0.0,
+                               [](double lhs, const Array* rhs) { return lhs + rhs->min(); });
+    }
+    if constexpr (std::is_same<BinaryOp, std::multiplies<double>>::value) {
+        // we need to be a bit careful with our signs
+        assert(operands_.size() > 0);
+
+        double lhs_low = operands_[0]->min();
+        double lhs_high = operands_[0]->max();
+
+        for (const Array* rhs_ptr : operands_ | std::views::drop(1)) {
+            double rhs_low = rhs_ptr->min();
+            double rhs_high = rhs_ptr->max();
+
+            double tmp = std::min({lhs_low * rhs_low, lhs_low * rhs_high, lhs_high * rhs_low,
+                                   lhs_high * rhs_high});
+
+            lhs_high = std::max({lhs_low * rhs_low, lhs_low * rhs_high, lhs_high * rhs_low,
+                                 lhs_high * rhs_high});
+
```

Comment:
I'm not sure this logic always holds, consider for example three operands that have these as the min/max:
```
[(1, 5), (-5, -1), (-1, 0)]
```

Here's how the loop would iterate
```
1. lhs_low, lhs_high = 1, 5
combos = (1 * -5, 1 * -1, 5 * -5, 5 * -1) = (-5, -1, -25, -5)
min(combos), max(combos) = -25, -1

2. lhs_low, lhs_high = -25, -1
combos = (-25 * -1, -25 * 0, -1 * -1, -1 * 0) = (25, 0, 1, 0)
min(combos), max(combos) = 1, 25

final lhs_low, lhs_high = 1, 25
```

So it would return 1, but the actual minimum is -25. Hope I got that right...

Reply:
I don't see how to get -25 out of your example. I can get it by using
```
[(1, 5), (-5, -1), (-1, 1)]
```
which does give
```
2. lhs_low, lhs_high = -25, -1
combos = (-25 * -1, -25 * 1, -1 * -1, -1 * 1) = (25, -25, 1, -1)
min(combos), max(combos) = -25, 25

final lhs_low, lhs_high = -25, 25
```

But there may be other counterexamples... Thinking.

Reply:
Good catch, I had tried to simplify my example and messed it up in the process. After writing some quick python to play around with this I'm now convinced the logic is sound.